### PR TITLE
Jsdoc tags

### PR DESCRIPTION
**Pull request by justinfagnani**
Adds support for `@customElement`, `@polymer`, `@mixinFunction`, `@mixinClass`, and `@appliesMixin`.

Removes the EditorService which got test times down to ~2s.

Refactors some JSDoc handling code to remove our custom Annotation/Tag interfaces in favor of just using Doctrine.

I first removed support for the legacy annotations, then added them back in another commit so it can be easily reverted when we remove support. The commit that adds them back duplicates tests for easy removal too.

It's probably easier to review this one commit-by-commit with the middle commit being the mst significant.

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

File: src/polymer/polymer-element.ts [hunk 1]
```diff
@@ -549,28 +547,28 @@ export function applyMixins(
...
         severity: Severity.ERROR,
-        code: 'mixes-reference-not-found',
+        code: 'applies-mixin-reference-not-found',
         sourceRange: scannedMixinReference.sourceRange!,
       });
       continue;
     } else if (mixins.size > 1) {
       element.warnings.push({
-        message: `@mixes reference, multiple mixins found ${mixinId}`,
+        message: `@appliesMixin reference, multiple mixins found ${mixinId}`,
```
Comment (rictic) [inline src/polymer/polymer-element.ts]: wording nit: `multiple mixins found with name ${mixinId}`

File: src/polymer/polymer-element.ts [hunk 2]
```diff
@@ -549,28 +547,28 @@ export function applyMixins(
...
+        code: 'applies-mixin-reference-multiple-found',
         sourceRange: scannedMixinReference.sourceRange!,
       });
       continue;
     }
     const mixin = mixins.values().next().value;
     if (!(mixin instanceof PolymerElementMixin)) {
       element.warnings.push({
-        message: `@mixes reference to a non-Mixin ${mixinId}`,
+        message: `@appliesMixin reference to a non-Mixin ${mixinId}`,
```
Comment (rictic) [inline src/polymer/polymer-element.ts]: non-Polymer mixin?

File: src/polymer/polymer2-element-scanner.ts [hunk 3]
```diff
@@ -133,6 +132,7 @@ export class Polymer2ElementScanner implements JavaScriptScanner {
     const className = element.clazz.namespacedName;
     let tagName: string|undefined = undefined;
     // TODO(rictic): support `@customElements explicit-tag-name` from jsdoc
+    // TODO(justinfagnani): do this in this PR?
```
Comment (rictic) [inline src/polymer/polymer2-element-scanner.ts]: recommend against doing this now, as this whole thing is getting a big rewrite in #625

File: src/polymer/pseudo-element-scanner.ts [hunk 4]
```diff
@@ -40,17 +40,18 @@ export class PseudoElementScanner implements HtmlScanner {
       if (dom5.isCommentNode(node) && node.data &&
           node.data.includes('@pseudoElement')) {
         const parsedJsdoc = jsdoc.parseJsdoc(node.data);
-        const pseudoTag =
-            jsdoc.getTag(parsedJsdoc, 'pseudoElement', 'name') || undefined;
-        if (pseudoTag) {
+        const pseudoTag = jsdoc.getTag(parsedJsdoc, 'pseudoElement');
+        console.log('pseudoTag', pseudoTag);
```
Comment (rictic) [inline src/polymer/pseudo-element-scanner.ts]: console.log

File: src/javascript/jsdoc.ts [hunk 5]
```diff
@@ -13,130 +13,65 @@
...
+  });
+  // Strip any leading and trailing newline characters in the
+  // description of multiline comments for readibility.
+  // TODO(rictic): figure out if we can trim() here or not. Something something
+  //     markdown?
+  const description = d.description && d.description.replace(/^\n+|\n+$/g, '');
+  return {description, tags: parseCustomTags(d.tags)};
+}
+
+const namedTags = new Set([
```
Comment (justinfagnani) [inline src/javascript/jsdoc.ts]: done

File: src/polymer/behavior-scanner.ts [hunk 6]
```diff
@@ -110,12 +110,12 @@ class BehaviorVisitor implements Visitor {
...
           message:
               `Unable to determine property name from expression of type ${
-                                                                           node.type
+        node.type
                                                                          }`,
           severity: Severity.WARNING,
           sourceRange: this.document.sourceRangeForNode(node)!
         });
-        continue;
+                                                                         continue;
```
Comment (justinfagnani) [inline src/polymer/behavior-scanner.ts]: trying, and I'll add a format commit

File: src/polymer/polymer-element.ts (hunk 2 above)
Comment (justinfagnani) [inline src/polymer/polymer-element.ts]: done

File: src/polymer/polymer-element.ts (hunk 1 above)
Comment (justinfagnani) [inline src/polymer/polymer-element.ts]: done

File: src/polymer/polymer2-element-scanner.ts (hunk 3 above)
Comment (justinfagnani) [inline src/polymer/polymer2-element-scanner.ts]: ack

File: src/polymer/pseudo-element-scanner.ts (hunk 4 above)
Comment (justinfagnani) [inline src/polymer/pseudo-element-scanner.ts]: done

**Comment by justinfagnani** [comment]
@rictic PTAL

**Comment by rictic** [comment]
Looks good, but travis is failing on `gulp build` is failing, maybe on json schema generation? Did any of the analysis format types change with this PR?

Other PRs aren't failing so I don't think this is an upstream thing

**Comment by justinfagnani** [comment]
@rictic Rebased and push -f'ed, but now I need to rebase again :(

**Pull request merged**